### PR TITLE
feat: emit inline frames

### PR DESCRIPTION
**Pull request by Gankra**
just messing around right now...

TODO:

* [x] human output cleanup
  * [x] fix the `:` in "Found by"
  * [x] add the source file/line to the end
  * [x] add the module to the start
    * ~~tbh I hate the `xul.dll!` prefix...~~
    * ~~[ ] check if [mozcrash.py](https://searchfox.org/mozilla-central/source/testing/mozbase/mozcrash/mozcrash/mozcrash.py) relies on it~~
* [x] tests
   * [x] add a new input minidump/sym that integration tests can slurp up to test output
* [x] docs
  * [x] update json-schema.md
  * [x] add doc-comments
  * [x] RELEASES.md
* [x] api
  * [x] look over the new APIs
  * [x] look over the new fields
  * [x] decide if we're ok with everyone using this API having to `.rev()` the inlinees (probably not, but it's not a terrible burden either)
      * [x] I was not, fixed 
* ~~[ ] builtin dump_syms support~~
  * ~~[ ] pass --inlines?~~
    * ~~configurable by a flag?~~

### CODE REVIEW

**Comment by Gankra** [comment]
current sample output:

<details><summary>output-human.txt</summary>

```
Crash reason:  EXC_BAD_ACCESS / KERN_INVALID_ADDRESS
Crash address: 0xffffffff80000042
Process uptime: 0 seconds

Thread 0 main (crashed)
 0  core::ptr::write_volatile
    Found by inlining
 1  crash-client!sadness_generator::raise_segfault [lib.rs : 133 + 0x0]
     rax = 0x000000010a000199    rdx = 0x0000700001f79000
     rcx = 0x000000010a00032c    rbx = 0x0000000000000004
     rsi = 0x00007fee420049c0    rdi = 0x00007fee42004a60
     rbp = 0x00007ffee5c02ce0    rsp = 0x00007ffee5c02ce0
      r8 = 0x00000000090008ff     r9 = 0x0000000000000003
     r10 = 0x0000700001f79000    r11 = 0x0000000000000246
     r12 = 0x0000000000000003    r13 = 0x00000000000000ff
     r14 = 0x0000000000000d03    r15 = 0x0000000200000000
     rip = 0x000000010a010c34
    Found by: given as instruction pointer in context
 2  crash_client::real_main::{{closure}}
    Found by inlining
 3  crash_client::real_main
    Found by inlining
 4  crash-client!crash_client::main [main.rs : 142 + 0x562]
     rbx = 0x0000000000000004    rbp = 0x00007ffee5c03350
     rsp = 0x00007ffee5c02cf0    r12 = 0x0000000000000003
     r13 = 0x00000000000000ff    r14 = 0x0000000000000d03
     r15 = 0x0000000200000000    rip = 0x000000010a00019e
    Found by: call frame info
 5  core::ops::function::FnOnce::call_once
    Found by inlining
 6  crash-client!std::sys_common::backtrace::__rust_begin_short_backtrace [backtrace.rs : 122 + 0x1]
     rbx = 0x00007fee41405ae0    rbp = 0x00007ffee5c03360
     rsp = 0x00007ffee5c03360    r12 = 0x00007ffee5405000
     r13 = 0x00007ffee5404000    r14 = 0x0000000000000001
     r15 = 0x000000010a0c01b8    rip = 0x000000010a001d86
    Found by: call frame info
 7  crash-client!std::rt::lang_start::{{closure}} [rt.rs : 145 + 0xb]
     rbx = 0x00007fee41405ae0    rbp = 0x00007ffee5c03370
     rsp = 0x00007ffee5c03370    r12 = 0x00007ffee5405000
     r13 = 0x00007ffee5404000    r14 = 0x0000000000000001
     r15 = 0x000000010a0c01b8    rip = 0x000000010a001a8c
    Found by: call frame info
 8  core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
    Found by inlining
 9  std::panicking::try::do_call
    Found by inlining
10  std::panicking::try
    Found by inlining
11  std::panic::catch_unwind
    Found by inlining
12  std::rt::lang_start_internal::{{closure}}
    Found by inlining
13  std::panicking::try::do_call
    Found by inlining
14  std::panicking::try
    Found by inlining
15  std::panic::catch_unwind
    Found by inlining
16  crash-client!std::rt::lang_start_internal [rt.rs : 128 + 0x3]
     rbx = 0x00007fee41405ae0    rbp = 0x00007ffee5c03480
     rsp = 0x00007ffee5c03380    r12 = 0x00007ffee5405000
     r13 = 0x00007ffee5404000    r14 = 0x0000000000000001
     r15 = 0x000000010a0c01b8    rip = 0x000000010a07f34e
    Found by: call frame info
17  crash-client!main + 0x28
     rbx = 0x0000000000000000    rbp = 0x00007ffee5c034a0
     rsp = 0x00007ffee5c03490    r12 = 0x0000000000000000
     r13 = 0x0000000000000000    r14 = 0x0000000000000000
     r15 = 0x0000000000000000    rip = 0x000000010a000379
    Found by: call frame info
18  libdyld.dylib + 0x15f3c
     rbx = 0x0000000000000000    rbp = 0x00007ffee5c034b0
     rsp = 0x00007ffee5c034b0    r12 = 0x0000000000000000
     r13 = 0x0000000000000000    r14 = 0x0000000000000000
     r15 = 0x0000000000000000    rip = 0x00007fff20329f3d
    Found by: call frame info
```

</details>

**Comment by Gankra** [comment]
see also:

* https://github.com/Gankra/minidump-pipeline/pull/3
* https://github.com/Gankra/minidump-debugger/pull/2

**Comment by Gankra** [comment]
lmfao sym files 
image

I guess I could try to trim the file to what's needed but that kinda sounds like a pain

File: minidump-processor/src/process_state.rs [hunk 1]
```diff
@@ -410,49 +458,86 @@ impl CallStack {
     ///
     /// This is very verbose, it implements the output format used by
     /// minidump_stackwalk.
+    /// This currently does not output inline frames.
```
Comment (Swatinem) [inline minidump-processor/src/process_state.rs:461]: well it does?

**Review by Swatinem — APPROVED** [review]
some minor comments, looks awesome otherwise

File: breakpad-symbols/src/sym_file/mod.rs [hunk 2]
```diff
@@ -361,10 +362,73 @@ impl SymbolFile {
...
+                    // There is an inline call at the address.
+                    // Enumerate all inlines at the address one by one by looking up
+                    // successively deeper call depths.
+                    // The call to `get_outermost_source_location` above looked up depth 0, so here
+                    // we start at depth 1.
+                    for depth in 1.. {
+                        match func.get_inlinee_at_depth(depth, addr) {
+                            Some((call_file_id, call_line, _address, next_inline_origin)) => {
+                                // We found another inline frame.
+                                let call_file = self.files.get(&call_file_id).map(Deref::deref);
```
[the remark this answers is not in the document]
Comment (Gankra) [reply breakpad-symbols/src/sym_file/mod.rs:411]: sadly it's an `Option<&String>` and we want `Option<&str>`. 

`as_deref()`  wants us to have an `Option<String>

**Pull request merged**